src/lava/lib/optimization/problems/problems.py

```diff
@@ -312,3 +295,123 @@ def get_constraint_biases(self) -> np.ndarray:
     @property
     def num_variables(self) -> int:
         return len(self._linear_offset)
+
+
+class IQP(OptimizationProblem):
+    """
+    Class to instantiate an Integer Quadratic Programming problem in the
+    standard form as:
+
+    .. math::               # TODO : Fix equation representation
+        min x^THx+c^Tx \\\\
+        Ax >= b \\\\
+        x >= 0 \\\\
+        x in Z
+    """
+
+    def __init__(
+        self,
+        H: np.ndarray,
+        c: np.ndarray,
+        A: np.ndarray,
+        b: np.ndarray,
+    ):
+        """
+        Constructor for the IQP class.
+
+        Parameters
+        ----------
+        H : 2-D np.array
+            Quadratic term of the cost function with integer coefficients.
+        c : 1-D np.array
+            Linear term of the cost function with integer coefficients.
+        A : 2-D or 1-D np.array
+            Equality constrainting hyperplanes.
+        b : 1-D np.array
+            Eqaulity constraints offsets.
+        """
+        super().__init__()
+        self._validate_input(H, c, A, b)
+        self._H = H
+        self._c = c
+        self._A = A
+        self._b = b
+
+        # TODO : Define domains
+        self._variables = DiscreteVariables(domains=[])
+        self._cost = Cost(*[coeff for coeff in [H, c] if coeff is not None])
+        # TODO : Define linear & box constraints
+        self._constraints = ArithmeticConstraints()
+
+    @property
+    def variables(self):
+        """Discrete variables over which the problem is specified."""
+        return self._variables
+
+    @property
+    def cost(self):
+        """Cost function."""
+        return self._cost
+
+    @property
+    def constraints(self):
+        """Specification of mutually allowed values between variables."""
+        return self._constraints
+
+    def _validate_input(
+        self, H: np.ndarray, c: np.ndarray, A: np.ndarray, b: np.ndarray
+    ) -> None:
+        for coeff in [H, c, A, b]:
+            if H is not None and not isinstance(coeff.flat[0], np.int32):
+                raise ValueError(
+                    f"Coefficients have to be np.int32 type, got {coeff.dtype}"
+                )
+        if H is not None and H.shape[0] != H.shape[1]:
+            raise ValueError
+        if c is not None and H is not None and c.shape[0] != H.shape[0]:
+            raise ValueError
+        if c is not None and A is not None and c.shape[0] != A.shape[1]:
+            raise ValueError
+        if A is not None and b is not None and A.shape[0] != b.shape[0]:
+            raise ValueError
+
+    def evaluate_cost(self, x: np.ndarray) -> int:
+        """Evaluate cost of provided solution."""
+        return self._cost(x)

```

Comment:
Also here: How can you call self._cost(x) if class Cost does not have a __call__?

Reply:
Cost is inheriting from CoefficientTensorsMixin that implements the call method, no need to reimplement it in Cost.